### PR TITLE
Fix: parsing of some date types

### PR DESCRIPTION
**Pull request by liss-h**

### CODE REVIEW

**Comment by liss-h** [comment]
Error messages are also not very good in some places. This PR improves this slightly to at least not be confusing but still far from good. Related #204

File: src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace rdf4cpp::rdf::datatypes::registry::util {
  * @return the resulting value
  * @throws std::runtime_error if the string cannot be parsed
  */
-template<std::integral I>
+template<std::integral I, ConstexprString datatype>
```
Comment (bigerl) [inline src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp:22]: Please document the template parameters here and in DateTimeUtils. Can be adressed together with making the `ConstexprString datatype` parameters more robust.

Comment (bigerl) [inline src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp:22]: `ConstexprString datatype` does not enforce in any way to be set to correct values only. It's very easy to set it to some random value. I have no simple idea at hand but a more robust approach would be preferable. This does not need to be addressed here but can be moved to a separate issue.

**Review by bigerl — APPROVED** [review]
Changes look good. I found a thing that would need improvement but that should be covered in another issue.

File: src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp (hunk 1 above)
Comment (liss-h) [reply src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp:22]: moved to error message issue

Comment (liss-h) [reply src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp:22]: moved to error message issue

**Pull request merged**